relay-server/src/actors/project_buffer.rs

```diff
@@ -380,6 +399,13 @@ impl BufferService {
         sender: &mpsc::UnboundedSender<ManagedEnvelope>,
     ) -> Result<(), QueueKey> {
         loop {
+            // By default this creates a prepared statement which is cached and re-used.
+            //
+            // Removing envelopes from the on-disk buffer in batches has following implicatations:
+            // 1. It is faster to delete from the DB in batches.
+            // 2. Make sure that if we panic and deleted envelopes cannot be read out fully, we do not lose all of them,
+            // but only one batch, and the rest of them will stay on disk for the next iteration
+            // to pick up.
             let mut envelopes = sqlx::query(
```

Comment:
From offline conversation: Provided this could be a DELTE without subquery if a specific feature is enabled, let's add that dependency ourselves, enable the feature, and leave a comment in cargo toml, why we have it.